### PR TITLE
Convert x-pack rolling upgrade tests to new test framework

### PR DESCRIPTION
**Pull request by mark-vieira**
Convert the `:x-pack:qa:rolling-upgrade` project to the new testing framework. There are a few idiosyncrasies with this project:

1. It's a BWC project with both Java and YAML test which is atypical. Had to tweak a few things for that to work.
2. There are a lot of Java test suites here. We might want to think about maybe consolidating some of them as there's a lot of overhead now that we're creating a lot more test clusters.
3. All the cluster configuration was shared before, so it was configured to the greatest common denominator of all the tests. We could probably simplify this in a lot of the individual tests.

### CODE REVIEW

**Comment by mark-vieira** [comment]
@elasticmachine update branch

File: x-pack/qa/rolling-upgrade/src/main/java/org/elasticsearch/upgrades/AbstractUpgradeTestCase.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        .keystore("xpack.watcher.encryption_key", Resource.fromClasspath("system_key"))
+        .user("test_user", "x-pack-test-password")
+        .user("non_operator", "x-pack-test-password", "superuser", false)
+        .feature(FeatureFlag.TIME_SERIES_MODE)
+        .build();
+
+    @ClassRule
+    public static TestRule ruleChain = RuleChain.outerRule(repo).around(cluster);
+
+    protected AbstractUpgradeTestCase(@Name("upgradedNodes") int upgradedNodes) {
```
Comment (breskeby) [inline x-pack/qa/rolling-upgrade/src/main/java/org/elasticsearch/upgrades/AbstractUpgradeTestCase.java:100]: is that @Name annotation used anywhere?

File: x-pack/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/ApiKeyBackwardsCompatibilityIT.java [hunk 2]
```diff
@@ -51,11 +54,19 @@ public class ApiKeyBackwardsCompatibilityIT extends AbstractUpgradeTestCase {
     private RestClient oldVersionClient = null;
     private RestClient newVersionClient = null;
 
-    public void testCreatingAndUpdatingApiKeys() throws Exception {
+    public ApiKeyBackwardsCompatibilityIT(@Name("upgradedNodes") int upgradedNodes) {
```
Comment (breskeby) [inline x-pack/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/ApiKeyBackwardsCompatibilityIT.java:57]: What is that annotation used for?

**Review by breskeby — APPROVED** [review]
lgtm. One remark / question.
I wonder what the best option would be to handle a huge set of test classes more efficient as you mentioned we now create way more cluster. Keeping those tests separated and organised as they are seems the best way to organise them. but could we change the framework to use a shared elasticsearch cluster maybe?

**Comment by mark-vieira** [comment]
> lgtm. One remark / question. I wonder what the best option would be to handle a huge set of test classes more efficient as you mentioned we now create way more cluster. Keeping those tests separated and organised as they are seems the best way to organise them. but could we change the framework to use a shared elasticsearch cluster maybe?

Yeah, there's no good way to do this with the current junit4 framework. We can't share a cluster across test classes in a way that works with these upgrade scenarios. We'd have a few more options with Junit5 I think but we're kind of limited with the randomized runner.

**Pull request closed without merging**